src/libraries/Microsoft.PowerFx.Connectors/Public/ConnectorTableValueWithServiceProvider.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Connectors
+{
+    public class ConnectorTableValueWithServiceProvider : ConnectorTableValue
+    {
+        internal IServiceProvider ServiceProvider { get; }
+
+        public ConnectorTableValueWithServiceProvider(ConnectorTableValue tableValue, IServiceProvider serviceProvider)
+            : base(tableValue._tabularService, (tableValue.IRContext.ResultType as ConnectorTableType).RecordType)
+        {
+            ServiceProvider = serviceProvider;
+        }
+
+        private IEnumerable<DValue<RecordValue>> _cachedRows;
```

Comment:
>IEnumerable [](http://example.com/codeflow?start=16&length=11)

can we change to something that includes length?

That ensures there isnt deferred network calls. 
All network calls should be in the duration of GetRows()

Reply:
I made the change but notice that CollectionTableValue.Rows is only IEnumerable<DValue<RecordValue>> and this leads to a ToArray() call in GetItemsAsync

Reply:
I don't see how paging could work here:
- the returned rows can't be a deferred object (ie, not IEnumerable) because network calls can't be deferred (they must use the current httpClient). 
- how would the page update the _cachedRows?

This could be an existing problem with our paging in dataverse.

Reply:
We'll deal with that later, not needed now